### PR TITLE
Generating .NET Track2 sdk FOR NETWORK

### PR DESCRIPTION
**Pull request by htippanaboya**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/network/Azure.ResourceManager.Network/src/Generated/BastionHostResource.cs [hunk 1]
```diff
@@ -221,32 +222,32 @@ public virtual ArmOperation Delete(WaitUntil waitUntil, CancellationToken cancel
...
-        /// <param name="data"> Parameters supplied to the create or update Bastion Host operation. </param>
+        /// <param name="networkTagsObject"> Parameters supplied to update BastionHost tags. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="data"/> is null. </exception>
-        public virtual async Task<ArmOperation<BastionHostResource>> UpdateAsync(WaitUntil waitUntil, BastionHostData data, CancellationToken cancellationToken = default)
+        /// <exception cref="ArgumentNullException"> <paramref name="networkTagsObject"/> is null. </exception>
+        public virtual async Task<ArmOperation<BastionHostResource>> UpdateAsync(WaitUntil waitUntil, NetworkTagsObject networkTagsObject, CancellationToken cancellationToken = default)
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/BastionHostResource.cs:241]: There are breaking changes in the interfaces.
- Add manual codes to provide backward compatibility, see compute for reference: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations
- By the way, there is already `SetTags` for this class: https://github.com/Azure/azure-sdk-for-net/blob/a9a133a89fdd2d3f4b40642776059b88a06c8345/sdk/network/Azure.ResourceManager.Network/src/Generated/BastionHostResource.cs#L413

File: sdk/network/Azure.ResourceManager.Network/src/Generated/CustomIPPrefixData.cs [hunk 2]
```diff
@@ -37,27 +37,37 @@ public CustomIPPrefixData()
...
+        internal CustomIPPrefixData(ResourceIdentifier id, string name, ResourceType? resourceType, AzureLocation? location, IDictionary<string, string> tags, ExtendedLocation extendedLocation, ETag? etag, IList<string> zones, string asn, string cidr, string signedMessage, string authorizationMessage, WritableSubResource customIPPrefixParent, IReadOnlyList<WritableSubResource> childCustomIPPrefixes, CommissionedState? commissionedState, bool? expressRouteAdvertise, Geo? geo, bool? noInternetAdvertise, CustomIPPrefixType? prefixType, IReadOnlyList<WritableSubResource> publicIPPrefixes, Guid? resourceGuid, string failedReason, NetworkProvisioningState? provisioningState) : base(id, name, resourceType, location, tags)
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/CustomIPPrefixData.cs:55]: There are breaking changes in the interfaces.
- Add manual codes to provide backward compatibility, see compute for reference: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations

**Comment by archerzz** [comment]
Some of the errors in pipeline will be resolved in https://github.com/Azure/azure-sdk-for-net/pull/35091

File: sdk/network/Azure.ResourceManager.Network/src/autorest.md [hunk 3]
```diff
@@ -396,54 +397,65 @@ These settings apply only when `--tag=package-track2-preview` is specified on th
...
-  - https://github.com/Azure/azure-rest-api-specs/blob/7384176da46425e7899708f263e0598b851358c2/specification/network/resource-manager/Microsoft.Network/stable/2021-02-01/virtualRouter.json
-  - https://github.com/Azure/azure-rest-api-specs/blob/7384176da46425e7899708f263e0598b851358c2/specification/network/resource-manager/Microsoft.Network/stable/2021-02-01/virtualWan.json
-#   - https://github.com/Azure/azure-rest-api-specs/blob/7384176da46425e7899708f263e0598b851358c2/specification/network/resource-manager/Microsoft.Network/stable/2021-02-01/vmssNetworkInterface.json
-#   - https://github.com/Azure/azure-rest-api-specs/blob/7384176da46425e7899708f263e0598b851358c2/specification/network/resource-manager/Microsoft.Network/stable/2021-02-01/vmssPublicIpAddress.json
-  - https://github.com/Azure/azure-rest-api-specs/blob/7384176da46425e7899708f263e0598b851358c2/specification/network/resource-manager/Microsoft.Network/stable/2021-02-01/webapplicationfirewall.json
+  - https://github.com/Azure/azure-rest-api-specs/blob/7f5ff3f2011f2f8019ff5db4b6cc00a128c01eb5/specification/network/resource-manager/Microsoft.Network/stable/2022-09-01/applicationGateway.json
```
Comment (ArthurMa1978) [inline sdk/network/Azure.ResourceManager.Network/src/autorest.md:400]: Please re-record all tests for the new version

File: sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs [hunk 4]
```diff
@@ -331,8 +453,8 @@ public partial class BastionHostResource : Azure.ResourceManager.ArmResource
...
-        public virtual Azure.ResourceManager.ArmOperation<Azure.ResourceManager.Network.BastionHostResource> Update(Azure.WaitUntil waitUntil, Azure.ResourceManager.Network.BastionHostData data, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.Network.BastionHostResource>> UpdateAsync(Azure.WaitUntil waitUntil, Azure.ResourceManager.Network.BastionHostData data, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.ResourceManager.ArmOperation<Azure.ResourceManager.Network.BastionHostResource> Update(Azure.WaitUntil waitUntil, Azure.ResourceManager.Network.Models.NetworkTagsObject networkTagsObject, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (ArthurMa1978) [inline sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs:456]: breaking change need to be fixed by custom code

File: sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs [hunk 5]
```diff
@@ -455,14 +619,19 @@ public partial class CustomIPPrefixCollection : Azure.ResourceManager.ArmCollect
     public partial class CustomIPPrefixData : Azure.ResourceManager.Network.Models.NetworkTrackedResourceData
     {
         public CustomIPPrefixData() { }
+        public string Asn { get { throw null; } set { } }
         public string AuthorizationMessage { get { throw null; } set { } }
-        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.Network.CustomIPPrefixData> ChildCustomIPPrefixes { get { throw null; } }
+        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.Resources.Models.WritableSubResource> ChildCustomIPPrefixes { get { throw null; } }
```
Comment (ArthurMa1978) [inline sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs:624]: breaking change need to be fixed by custom code

File: sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs [hunk 6]
```diff
@@ -455,14 +619,19 @@ public partial class CustomIPPrefixCollection : Azure.ResourceManager.ArmCollect
...
         public string Cidr { get { throw null; } set { } }
         public Azure.ResourceManager.Network.Models.CommissionedState? CommissionedState { get { throw null; } set { } }
-        public Azure.ResourceManager.Network.CustomIPPrefixData CustomIPPrefixParent { get { throw null; } set { } }
+        public Azure.Core.ResourceIdentifier CustomIPPrefixParentId { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs:627]: breaking change need to be fixed by custom code

File: sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs [hunk 7]
```diff
@@ -503,9 +672,7 @@ public partial class DdosCustomPolicyData : Azure.ResourceManager.Network.Models
     {
         public DdosCustomPolicyData() { }
         public Azure.ETag? ETag { get { throw null; } }
-        public System.Collections.Generic.IList<Azure.ResourceManager.Network.Models.ProtocolCustomSettings> ProtocolCustomSettings { get { throw null; } }
```
Comment (ArthurMa1978) [inline sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs:506]: breaking change need to be fixed by custom code

File: sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs [hunk 8]
```diff
@@ -1206,18 +1458,20 @@ public partial class FirewallPolicyCollection : Azure.ResourceManager.ArmCollect
...
         public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.Resources.Models.WritableSubResource> Firewalls { get { throw null; } }
         public Azure.ResourceManager.Models.ManagedServiceIdentity Identity { get { throw null; } set { } }
         public Azure.ResourceManager.Network.Models.FirewallPolicyInsights Insights { get { throw null; } set { } }
         public Azure.ResourceManager.Network.Models.FirewallPolicyIntrusionDetection IntrusionDetection { get { throw null; } set { } }
         public Azure.ResourceManager.Network.Models.NetworkProvisioningState? ProvisioningState { get { throw null; } }
         public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.Resources.Models.WritableSubResource> RuleCollectionGroups { get { throw null; } }
         public Azure.ResourceManager.Network.Models.FirewallPolicySkuTier? SkuTier { get { throw null; } set { } }
-        public System.Collections.Generic.IList<string> SnatPrivateRanges { get { throw null; } }
+        public Azure.ResourceManager.Network.Models.FirewallPolicySnat Snat { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs:1474]: breaking change need to be fixed by custom code

File: sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs [hunk 9]
```diff
@@ -1235,15 +1489,20 @@ public partial class FirewallPolicyResource : Azure.ResourceManager.ArmResource
...
-        public virtual Azure.ResourceManager.ArmOperation<Azure.ResourceManager.Network.FirewallPolicyResource> Update(Azure.WaitUntil waitUntil, Azure.ResourceManager.Network.FirewallPolicyData data, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.Network.FirewallPolicyResource>> UpdateAsync(Azure.WaitUntil waitUntil, Azure.ResourceManager.Network.FirewallPolicyData data, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response<Azure.ResourceManager.Network.FirewallPolicyResource> Update(Azure.ResourceManager.Network.Models.NetworkTagsObject networkTagsObject, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (ArthurMa1978) [inline sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs:1504]: breaking change need to be fixed by custom code

File: sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs [hunk 10]
```diff
@@ -8977,13 +10900,49 @@ public partial class ServiceTagsListResult : Azure.ResourceManager.Models.Resour
         public string NextLink { get { throw null; } }
         public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.Network.Models.ServiceTagInformation> Values { get { throw null; } }
     }
+    public partial class SignatureOverridesFilterValuesQuery
+    {
+        public SignatureOverridesFilterValuesQuery() { }
+        public string FilterName { get { throw null; } set { } }
+    }
+    public partial class SignatureOverridesFilterValuesResponse
```
Comment (ArthurMa1978) [inline sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs:10908]: rename to `SignatureOverridesFilterValuesResult`

File: sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs [hunk 11]
```diff
@@ -8977,13 +10900,49 @@ public partial class ServiceTagsListResult : Azure.ResourceManager.Models.Resour
         public string NextLink { get { throw null; } }
         public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.Network.Models.ServiceTagInformation> Values { get { throw null; } }
     }
+    public partial class SignatureOverridesFilterValuesQuery
```
Comment (ArthurMa1978) [inline sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs:10903]: rename to `SignatureOverridesFilterValuesContent`

Comment (htippanaboya) [reply sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs]: Made the change in this commit https://github.com/Azure/azure-sdk-for-net/pull/34959/commits/70d74ef481f2b2243288810f62a0e6e891bef8ae

File: sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs (hunk 10 above)
Comment (htippanaboya) [reply sdk/network/Azure.ResourceManager.Network/api/Azure.ResourceManager.Network.netstandard2.0.cs:10908]: Made the change in this commit https://github.com/Azure/azure-sdk-for-net/pull/34959/commits/70d74ef481f2b2243288810f62a0e6e891bef8ae

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**